### PR TITLE
Update yaml-language-server to be compatible with vscode-json-languag…

### PR DESCRIPTION
**Pull request by JPinkney**
…eservice 3.10.0

Signed-off-by: Josh Pinkney <joshpinkney@gmail.com>

### What does this PR do?
This PR updates yaml ls to be compatible with the newest version of vscode-json-languageservice.
Notable changes:
- Removes completion resolve since it doesn't actually do anything
- Maintains findDefinition when using this project as a library but makes it return [], just for compatibility reasons (vscode-json-languageservice does this as well)
- Implements findLinks based off of the newest vscode-json-languageservice which returns all document links in the document

### What issues does this PR fix or reference?
https://github.com/redhat-developer/yaml-language-server/issues/345

### Is it tested? How?

npm run test

### CODE REVIEW

File: src/languageservice/services/yamlCompletion.ts [hunk 1]
```diff
@@ -225,19 +211,6 @@ export class YAMLCompletion extends JSONCompletion {
         }
 
         const location = Parser.getNodePath(node);
```
Comment (joshuawilson) [inline src/languageservice/services/yamlCompletion.ts:213]: Do you need to leave `location` here now that you have removed the `contributions.forEach`?

File: src/languageservice/services/yamlLinks.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+export function findLinks(document: TextDocument): Thenable<DocumentLink[]> {
+  const doc = parseYAML(document.getText());
+  // Find links across all YAML Documents then report them back once finished
+  const linkPromises = [];
+  for (const yamlDoc of doc.documents) {
+    linkPromises.push(JSONFindLinks(document, yamlDoc));
+  }
+  // Wait for all the promises to return and then flatten them into one DocumentLink array
+  return Promise.all(linkPromises).then(yamlLinkArray => [].concat(...yamlLinkArray));
```
Comment (joshuawilson) [inline src/languageservice/services/yamlLinks.ts:18]: I think the build is complaining that it wants it to look like this.
`return Promise.all(linkPromises).then((yamlLinkArray) => [].concat(...yamlLinkArray));`
with the inner function parameter to be wrapped in parens.

File: src/languageservice/yamlLanguageService.ts [hunk 3]
```diff
@@ -24,8 +23,8 @@ import { YAMLHover } from './services/yamlHover';
 import { YAMLValidation } from './services/yamlValidation';
 import { YAMLFormatter } from './services/yamlFormatter';
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-import { getLanguageService as getJSONLanguageService, JSONWorkerContribution } from 'vscode-json-languageservice';
-import { findDefinition } from './services/yamlDefinition';
+import { getLanguageService as getJSONLanguageService, JSONWorkerContribution, JSONDocument, DefinitionLink } from 'vscode-json-languageservice';
```
Comment (joshuawilson) [inline src/languageservice/yamlLanguageService.ts:26]: This just needs to be made multi-line.

File: test/findLinks.test.ts [hunk 4]
```diff
@@ -5,44 +5,35 @@
...
-              line: 2,
-              character: 4,
-            },
-            end: {
-              line: 2,
-              character: 16,
+              character: 29,
             },
           });
+          assert.deepEqual(results[0].target, "file://~/Desktop/vscode-k8s/test.yaml#3,5");
```
Comment (joshuawilson) [inline test/findLinks.test.ts:36]: use single quotes

**Review by joshuawilson — APPROVED** [review]

**Comment by marcusramberg** [comment]
Any plans for a new release to get this fix out?

**Comment by evidolob** [comment]
@marcusramberg I will start bugfix release in a few hours.

**Comment by JPinkney** [comment]
Should we do bugfix release or full release? Since https://github.com/redhat-developer/yaml-language-server/commit/4708ed0df3fab7135d88ba4c03a09e74c140d42c and https://github.com/redhat-developer/yaml-language-server/commit/6024f9ab65b171f1c1fea7dd9696ec028239d984 are new features

**Comment by evidolob** [comment]
Then full release.

**Pull request merged**